src/lib.rs

```diff
@@ -292,6 +319,27 @@ impl OpenatDirExt for openat::Dir {
         impl_remove_all(self, p)
     }
 
+    fn update_timestamps<P: openat::AsPath>(&self, p: P) -> io::Result<()> {
+        use nix::sys::stat::{utimensat, UtimensatFlags};
+        use nix::sys::time::TimeSpec;
+
+        let path = p
+            .to_path()
+            .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "null byte in path"))?;
+        let now = TimeSpec::from(libc::timespec {
+            tv_nsec: libc::UTIME_NOW,
+            tv_sec: 0,
```

Comment:
Shouldn't this be https://docs.rs/libc/0.2.94/libc/constant.UTIME_NOW.html ?

Reply:
According to the [manpage]() in this case the second field is ignored and only `tv_nsec` matters:

       If the tv_nsec field of one of the timespec structures has the
       special value UTIME_NOW, then the corresponding file timestamp is
       set to the current time.  If the tv_nsec field of one of the
       timespec structures has the special value UTIME_OMIT, then the
       corresponding file timestamp is left unchanged.  In both of these
       cases, the value of the corresponding tv_sec field is ignored.

I double-checked the [kernel logic](https://github.com/torvalds/linux/blob/v5.12/fs/utimes.c#L19) and indeed it takes some shortcuts based on `tv_nsec` field alone.